ui/test/nuxeo-layout.test.js

```diff
@@ -27,57 +35,161 @@ const base = url.substring(0, url.lastIndexOf('/'));
-    const layout = await fixture(html`
-      <nuxeo-layout href="${base}/dummy-layout.html" model='{"text": "invalid", "required": "foo"}'></nuxeo-layout>
-    `);
-    const validity = await layout.validate();
-    expect(validity).to.equal(false);
+      layout.href = `${base}/layouts/document/test/nuxeo-test-view-layout.html`;
+      await flush();
+
+      if (layout.element.tagName === 'DUMMY-LAYOUT') {
+        await waitChanged(layout, 'element');
+      }
+
+      expect(layout.element.tagName).to.equal('NUXEO-TEST-VIEW-LAYOUT');
+      expect(layout.element.shadowRoot.querySelector('h1')).to.exist;
+      expect(layout.element.shadowRoot.querySelector('span')).to.not.exist;
+      expect(layout.element.shadowRoot.querySelector('nuxeo-input')).to.not.exist;
+    });
   });
 
-  test('layout invalid field invalid', async () => {
-    const layout = await fixture(html`
-      <nuxeo-layout href="${base}/dummy-layout.html" model='{"text": "invalid"}'></nuxeo-layout>
-    `);
-    const validity = await layout.validate();
-    expect(validity).to.equal(false);
+  suite('Validation', () => {
+    test('Should validate the layout when fields are valid', async () => {
+      const layout = await fixture(html`
+        <nuxeo-layout href="${base}/layouts/dummy-layout.html" model='{"text": "valid", "data": "foo"}'></nuxeo-layout>
+      `);
+      await flush();
+      await layoutLoad(layout);
+
+      const nuxeoInput = layout.element.shadowRoot.querySelector('nuxeo-input');
+      sinon.spy(nuxeoInput, 'validate');
+      sinon.spy(layout.element, 'validate');
+      sinon.spy(layout, 'validate');
+      const validity = await layout.validate();
+      expect(validity).to.be.true;
+      expect(nuxeoInput.invalid).to.be.false;
+      expect(nuxeoInput.validate.calledOnce).to.be.true;
+      expect(layout.element.validate.calledOnce).to.be.true;
+      expect(layout.validate.calledOnce).to.be.true;
+    });
+
+    test('Should invalidate the layout when native validation is invalid', async () => {
+      const layout = await fixture(html`
+        <nuxeo-layout href="${base}/layouts/dummy-layout.html" model=""></nuxeo-layout>
+      `);
+      await flush();
+      await layoutLoad(layout);
+
+      const nuxeoInput = layout.element.shadowRoot.querySelector('nuxeo-input');
+      const validity = await layout.validate();
+      expect(validity).to.be.false;
+      expect(nuxeoInput.invalid).to.be.true;
+    });
+
+    test('Should not run custom validation when native validation fails', async () => {
+      const layout = await fixture(html`
+        <nuxeo-layout href="${base}/layouts/dummy-layout.html" model=""></nuxeo-layout>
+      `);
+      await flush();
+      await layoutLoad(layout);
+
+      const nuxeoInput = layout.element.shadowRoot.querySelector('nuxeo-input');
+      sinon.spy(nuxeoInput, 'validate');
+      sinon.spy(layout.element, 'validate');
+      sinon.spy(layout, 'validate');
```

Comment:
Same here. It is not useful to declare a spy of something that we call directly.